loki-logback-appender/src/main/java/com/github/loki4j/logback/Loki4jAppender.java

```diff
@@ -231,10 +240,20 @@ public void setSendQueueMaxBytes(long sendQueueMaxBytes) {
     }
 
     public void setMaxRetries(int maxRetries) {
-        this.maxRetries = maxRetries;
+        if (maxRetries > 0 && maxRetries <= MAX_RETRIES) {
+            this.maxRetries = maxRetries;
+        } else {
+            this.maxRetries = MAX_RETRIES;
+            addError("Invalid value for `maxRetries`, using " + MAX_RETRIES + " instead.");
+        }
     }
     public void setRetryTimeoutMs(long retryTimeoutMs) {
-        this.retryTimeoutMs = retryTimeoutMs;
+        if (retryTimeoutMs > 0 && retryTimeoutMs <= MAX_RETRY_TIMEOUT_MS) {
```

Comment:
Is there anything nontrivial? Otherwise, let's remove this check.

Reply:
Same as with retries, if you set a high enough number here, with say 30 retries then you overflow the `long`, this max limit together with the max retries enforces that the retry wait is never overflown.